packages/webapp/src/containers/Finances/util.js

```diff
@@ -116,3 +121,108 @@ export function calcActualRevenue(sales, startDate, endDate, revenueTypes) {
 export const getRevenueFormType = (revenueType) => {
   return revenueType?.crop_generated ? revenueFormTypes.CROP_SALE : revenueFormTypes.GENERAL;
 };
+
+export const mapTasksToLabourItems = (tasks, taskTypes, users) => {
+  const groupingOptions = [
+    {
+      key: 'assignee_user_id',
+      label: LABOUR_ITEMS_GROUPING_OPTIONS.EMPLOYEE,
+      taskObject: (task, groupKey) => {
+        const assignee = users.find((user) => user.user_id == groupKey);
+        return {
+          ...task,
+          employee: `${assignee.first_name} ${assignee.last_name.substring(0, 1).toUpperCase()}.`,
+        };
+      },
+    },
+    {
+      key: 'task_type_id',
+      label: LABOUR_ITEMS_GROUPING_OPTIONS.TASK_TYPE,
+      taskObject: (task, groupKey) => {
+        const taskType = taskTypes.find((taskType) => taskType.task_type_id == groupKey);
+        return {
+          ...task,
+          taskType: i18n.t(`task:${taskType.task_translation_key}`),
+        };
+      },
+    },
+  ];
+  const labourItemGroups = {};
+  groupingOptions.forEach((option) => {
+    const groupedTasks = groupBy(tasks, (task) => task[option.key]);
+    const items = Object.keys(groupedTasks).map((groupKey) => {
+      const tasksInGroup = groupedTasks[groupKey].map((task) => {
+        const minutes = parseInt(task.duration, 10);
+        const hours = roundToTwoDecimal(minutes / 60);
+        const rate = roundToTwoDecimal(task.wage_at_moment);
+        const labourCost = roundToTwoDecimal(rate * hours);
+        return {
+          ...task,
+          time: hours,
+          labourCost,
+        };
+      });
+      const time = tasksInGroup.reduce((sum, task) => sum + task.time, 0);
+      const labourCost = tasksInGroup.reduce((sum, task) => sum + task.labourCost, 0);
+
+      return option.taskObject(
+        {
+          time: roundToTwoDecimal(time),
+          labourCost,
+        },
+        groupKey,
+      );
+    });
+    labourItemGroups[option.label] = items;
+  });
+
+  return labourItemGroups;
+};
+
+export const mapSalesToRevenueItems = (sales, revenueTypes, cropVarieties) => {
+  const revenueItems = sales.map((sale) => {
+    const revenueType = revenueTypes.find(
+      (revenueType) => revenueType.revenue_type_id === sale.revenue_type_id,
+    );
+    if (getRevenueFormType(revenueType) === revenueFormTypes.CROP_SALE) {
+      const quantityUnit = getMassUnit();
+      const cropVarietySale = sale.crop_variety_sale;
+      return {
+        sale,
+        totalAmount: cropVarietySale.reduce((total, sale) => total + sale.sale_value, 0),
+        financeItemsProps: cropVarietySale.map((cvs) => {
```

Comment:
Do we expect to use `financeItemsProps` for the desktop view as well in the future?
If so, I think:
- title and subTitle should be `crop` and `quantity` (they have dedicated columns in the desktop view) 
- dealing with numbers as string is a little tricky when sorting, so I think adding `quantityUnit` later is easier.

Reply:
Yeah I think you're right, it'd be good to have this be adaptable to the desktop view too so it's less rework in the future. Will update this!